server/openapi_metric.go

```diff
@@ -0,0 +1,41 @@
+package server
+
+import (
+	"strconv"
+	"time"
+
+	metricAPI "github.com/bytebase/bytebase/metric"
+	"github.com/bytebase/bytebase/plugin/metric"
+	"github.com/labstack/echo/v4"
+)
+
+func openAPIMetricMiddleware(s *Server, next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) (err error) {
+		start := time.Now()
+
+		defer func() {
+			duration := time.Since(start)
+			requestMethod := c.Request().Method
+			requestPath := c.Path()
+			responseCode := c.Response().Status
+
+			if s.MetricReporter != nil {
+				s.MetricReporter.report(&metric.Metric{
+					Name:  metricAPI.OpenAPIMetricName,
+					Value: 1,
+					Labels: map[string]string{
+						"latency_ms":     strconv.FormatInt(duration.Milliseconds(), 10),
```

Comment:
Shall we use nanoseconds for API/storage?

https://github.com/bytebase/bytebase/blob/main/Dockerfile

Reply:
LGTM, changed to nanoseconds